com.unity.render-pipelines.high-definition/Runtime/Material/Fabric/FabricPathTracing.hlsl

```diff
@@ -0,0 +1,280 @@
+#include "Packages/com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/PathTracing/Shaders/PathTracingIntersection.hlsl"
+#include "Packages/com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/PathTracing/Shaders/PathTracingMaterial.hlsl"
+#include "Packages/com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/PathTracing/Shaders/PathTracingBSDF.hlsl"
+
+// Fabric Material Data:
+//
+// Cotton/Wool mode:
+// bsdfWeight0  Diffuse BRDF
+// bsdfWeight1  Sheen BRDF
+// bsdfWeight2  Diffuse BTDF
+//
+// Silk mode:
+// bsdfWeight0  Diffuse BRDF
+// bsdfWeight1  Spec GGX BRDF
+// bsdfWeight2  Diffuse BTDF
+
+void ProcessBSDFData(PathIntersection pathIntersection, BuiltinData builtinData, inout BSDFData bsdfData)
+{
+    // Adjust roughness to reduce fireflies
+    bsdfData.roughnessT = max(pathIntersection.maxRoughness, bsdfData.roughnessT);
+    bsdfData.roughnessB = max(pathIntersection.maxRoughness, bsdfData.roughnessB);
+
+    if (HasFlag(bsdfData.materialFeatures, MATERIALFEATUREFLAGS_FABRIC_COTTON_WOOL))
+    {
+        // This is hacky, but applied to match the raster implementation (Fabric.hlsl)
+        bsdfData.diffuseColor *= FabricLambertNoPI(bsdfData.roughnessT);
+    }
+}
+
+bool CreateMaterialData(PathIntersection pathIntersection, BuiltinData builtinData, BSDFData bsdfData, inout float3 shadingPosition, inout float theSample, out MaterialData mtlData)
+{
+    // Alter values in the material's bsdfData struct, to better suit path tracing
+    mtlData.bsdfData = bsdfData;
+    ProcessBSDFData(pathIntersection, builtinData, mtlData.bsdfData);
+
+    mtlData.bsdfWeight = 0.0;
+    mtlData.V = -WorldRayDirection();
+
+    if (!IsAbove(mtlData))
+        return false;
+
+    mtlData.bsdfWeight[0] = Luminance(mtlData.bsdfData.diffuseColor) * mtlData.bsdfData.ambientOcclusion;
+
+    // If N.V < 0 (can happen with normal mapping) we want to avoid spec sampling
+    float NdotV = dot(mtlData.bsdfData.normalWS, mtlData.V);
+    if (NdotV > 0.001)
+    {
+        mtlData.bsdfWeight[1] = HasFlag(mtlData.bsdfData.materialFeatures, MATERIALFEATUREFLAGS_FABRIC_COTTON_WOOL) ?
+            0.1 * Luminance(mtlData.bsdfData.fresnel0) : Luminance(F_Schlick(mtlData.bsdfData.fresnel0, NdotV));
```

Comment:
why is there the * 0.1 here? maybe add some comment?

Reply:
I added some comments.